### ISSUE TITLE
question / Marker popup to display a filename

### ISSUE THREAD

**Issue by NO-MAN-CAT** [nemethviktor/GeoTagNinja#103]
**Is your feature request related to a problem? Please describe.**
Right now, I can select a bunch of photos and display them on a map. When I click on the geo marker on the map, **I would like to see the filename of the photo**, not only the coordinates.

**Describe the solution you'd like**
![image](https://github.com/nemethviktor/GeoTagNinja/assets/142752690/b13279cd-b68f-4764-97dc-8e96ce615c15)

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Comment by NO-MAN-CAT** [comment]
I absolutely don't mind if you push back on this suggestion. 😃  
I noticed the "issue" and proposed a solution for it. 
I agree with you that not everyone uses the program the same way I do. 

I would also like to suggest another small "feature" where a line is drawn on the map connecting the selected photos - for example, if the user selects 3 photos, in addition to markers on the map, I would like to see a line to understand which marker corresponds to which photo.

Here is an example:
![image](https://github.com/nemethviktor/GeoTagNinja/assets/142752690/c0c78873-8b3f-4972-9e04-1d5e2e5cb175)

**Comment by nemethviktor** [comment]
Hi, 

So going back to your original request I had a think about this and it can't be done because of a recursion issue. I'll try to explain why.
TLDR is that we're displaying the map _first_ and _then_ interacting with a click event so we can't have the click event influence the map display because that comes _after_.
The longer version is: I don't know how technical you are but...the map you see in the app is actually a html (`Resources\map.html`) file that gets pulled in and some elements in there are replaced or processed during runtime. Basically map you see in the app is a browser add-in (Edge) that instead of reading a "real" website reads the contents of the html and does some runtime modifications on the fly. This is controlled by `FrmMainApp.Request_Map_NavigateGo()`  - when the user does "map related stuff" the html script gets generated _first_ and _then_ it's shown (there are a whole bunch of `.Replace` in the code so in the html file you'll see stuff like `apikey: "yourApiKey", nearby: {lat: replaceLat,lng: replaceLng}`, which are handled on the fly by injection. Now `lat/long` are generally handled by the underlying `leaflet` framework, an opensource third-party stuff I'm using and from a technical perspective it's an invisible layer of sorts. 

If you look in the file there's some stuff the `map.html` itself:
```
map.on('click', function onMapClick(e) {
            //marker
            //    .setLatLng(e.latlng);

            popup
                .setLatLng(e.latlng)
                .setContent(e.latlng.toString())
                .openOn(map);

            window.chrome.webview.postMessage({ "lat": e.latlng.lat, "lng": e.latlng.lng, "isDragged": false });

        });
```

this means that when the user clicks on a point on the map
1) a popup is shown and 
2)  the coordinates are sent back to the app for further processing (last long line of code)

here's the issue: what you're basically asking is that we send a message back, _then_ we collate the files with those coordinates and _then_ we show the popup with the extended info. the trouble is that it wouldn't be sufficient to move the last line to precede the `popup` block because regardless of the order the whole thing is "one block/unit" of actions that can't be split, therefore one step would be relying on another step that's not possible to pull in the same order, so apologies, the original one won't be done as it's not possible.

**Comment by nemethviktor** [comment]
For the other one I've opened a separate ticket #104 -- will reply there.

**Comment by nemethviktor** [comment]
Closing as not possible due to recursion in request
